src/main/java/de/cronn/validation_files_diff/helper/PsiElementValidationFileFinder.java

```diff
@@ -0,0 +1,101 @@
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.LocalFileSystem;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.openapi.vfs.VirtualFileManager;
+import com.intellij.psi.PsiClass;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiMethod;
+import de.cronn.validation_files_diff.ValidationDiffProjectOptionsProvider;
+import org.jetbrains.annotations.Nullable;
+
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class PsiElementValidationFileFinder {
+
+    private final Project project;
+    private final Module module;
+    private final PsiElement element;
+
+    public PsiElementValidationFileFinder(PsiElement element, Project project, Module module) {
+        this.project = project;
+        this.module = module;
+        this.element = element;
+    }
+
+    public static PsiElementValidationFileFinder of(PsiElement element) {
+		Project project = element.getProject();
+		Module currentModule = ModuleUtil.findModuleForFile(element.getContainingFile());
+        return new PsiElementValidationFileFinder(element, project, currentModule);
+    }
+
+    public boolean hasCorrespondingValidationFiles() {
+        return !findCorrespondingValidationFiles().isEmpty();
+    }
+
+    public List<VirtualFile> findCorrespondingValidationFiles() {
+        VirtualFile moduleRoot = getModuleRoot();
+        if (moduleRoot == null) {
+            return Collections.emptyList();
+        }
+
+        String validationFilePrefix = parseValidationFilePrefix();
+
+        return getValidationFileRelatedDirectories()
+                .map(moduleRoot::findFileByRelativePath)
+                .filter(Objects::nonNull)
+                .map(VirtualFile::getChildren)
+                .flatMap(Arrays::stream)
+                .filter(validationFile -> fileNameStartsWith(validationFile, validationFilePrefix))
+                .collect(Collectors.toList());
+    }
+
+    private Stream<String> getValidationFileRelatedDirectories() {
+        ValidationDiffProjectOptionsProvider options = ValidationDiffProjectOptionsProvider.getInstance(project);
+        return Stream.of(options.getRelativeOutputDirPath(), options.getRelativeValidationDirPath(), options.getRelativeTempDirPath());
+    }
+
+    private boolean fileNameStartsWith(VirtualFile file, String prefix) {
+        return file.getName().startsWith(prefix);
+    }
+
+    private String parseValidationFilePrefix() {
+        if (element instanceof PsiMethod) {
+            PsiMethod psiMethod = (PsiMethod) element;
+            PsiClass psiClass = psiMethod.getContainingClass();
+            return PsiTestNameUtils.getTestName(psiClass, psiMethod);
+        }
+
+        if (element instanceof PsiClass) {
+            return PsiTestNameUtils.getTestClassName((PsiClass) element);
```

Comment:
Use pattern variable

```suggestion
        if (element instanceof PsiMethod psiMethod) {
			PsiClass psiClass = psiMethod.getContainingClass();
            return PsiTestNameUtils.getTestName(psiClass, psiMethod);
        }

        if (element instanceof PsiClass psiClass) {
            return PsiTestNameUtils.getTestClassName(psiClass);
```

Reply:
Done 8438ea52268c114eb81560ab254d4c9a81bd614d